packages/docregistry/src/context.ts

```diff
@@ -362,12 +362,31 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
     }
     let oldPath = change.oldValue && change.oldValue.path;
     let newPath = change.newValue && change.newValue.path;
-    if (newPath && oldPath === this._path) {
+
+    if (newPath && this._path.indexOf(oldPath) === 0) {
+      let changeModel = change.newValue;
+      // When folder name changed, `oldPath` is `foo`, `newPath` is `bar` and `this._path` is `foo/test`,
+      // we should update `foo/test` to `bar/test` as well
+      if (oldPath !== this._path) {
+        newPath = this._path.replace(new RegExp(`^${oldPath}`), newPath);
+        oldPath = this._path;
+        // Update client file model from folder change
+        changeModel = {
+          last_modified: change.newValue.created,
+          path: newPath
+        };
+      }
       this.session.setPath(newPath);
+      const updateModel = {
+        ...this._contentsModel,
+        last_modified: changeModel.last_modified,
+        name: changeModel.name || this._contentsModel.name,
```

Comment:
If you also set `name: this._contentsModel.name` on `changeModel` above, I think you can also do destructuring for these last three properties (`...changeModel`) and it might be a bit easier to read.

Reply:
Great, that would be nice.